### PR TITLE
posix shm, can not unmap shmem_ptr until encoder is done with it

### PR DESCRIPTION
**Pull request by jsorg71**
Fix issue where main thread unmap shared memory before worker thread is done with it.

### CODE REVIEW

File: xup/xup.c [hunk 1]
```diff
@@ -1386,11 +1386,14 @@ process_server_paint_rect_shmfd(struct mod *amod, struct stream *s)
             {
                 bmpdata = (char *)shmem_ptr;
                 bmpdata += shmem_offset;
+                /* we give up ownership of shmem_ptr
+                   will get cleaned up in server_paint_rects_ex or
+                   xrdp_mm_process_enc_done(rfx, gfx) */
```
Comment (matt335672) [inline xup/xup.c:1391]: Nice comment - makes it clear what's going on.

**Comment by matt335672** [comment]
LGTM

@Nexarian - this is more your area than mine. Amy comment?

**Comment by Nexarian** [comment]
I actually just verified that this works on mainline_merge with x264! :)

Now to get it working with the helper.

**Comment by jsorg71** [comment]
I think I'll merge the xorgxrdp part now.  It's good we still keep the sysV shm messages in xrdp(xup) so a newer xrdp will work with older xorgxrdp.  The other way around won't work but we could bump the header version so xorgxrdp won't compile.  We also have a version message in the API for run time check.  xup sends a version number to xorgxrdp, so far it's always been 0.0.0.1.

**Pull request merged**